mirrord/kube/src/api/kubernetes.rs

```diff
@@ -57,6 +57,23 @@ impl KubernetesAPI {
             target,
         }
     }
+
+    pub async fn detect_openshift<P>(&self, progress: &P) -> ()
```

Comment:
Do we need to return the `()`?

Have not looked into the progress stuff, so my bad if this is a requirement.